### ISSUE TITLE
Memory leak during tests

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#5647]
Successful tests often end with this warning: `Running coverage on untested files...A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --runInBand --detectOpenHandles to find leaks.`

This is on OSX but may be platform-independent.

**Comment by viceice** [comment]
Yes, I've seen them too. `--detectOpenHandles` never find something relevant. 🤔

**Comment by rarkins** [comment]
Blocked by https://github.com/facebook/jest/issues/7874

**Comment by Fyko** [comment]
I was able to resolve this in one of my projects by using `jest-circus` as my test runner:
```ts
import type { Config } from '@jest/types';

// eslint-disable-next-line @typescript-eslint/require-await
export default async (): Promise<Config.InitialOptions> => ({
	preset: 'ts-jest',
	testEnvironment: 'node',
	testRunner: 'jest-circus/runner',
        // other options removed for brevity
});

```

**Comment by viceice** [comment]
We already use `jest-circus`

**Comment by viceice** [comment]
i think we can close this, as our main memory leaker was `nock` ? https://github.com/nock/nock/issues/2057

latest nock fixes from #10451 
![image](https://user-images.githubusercontent.com/1798109/122208533-b6651100-cea3-11eb-9d22-9efd4aae308a.png)

before #10451 
![image](https://user-images.githubusercontent.com/1798109/122208680-e44a5580-cea3-11eb-957f-8e39d26a70e2.png)
